packages/babel-helper-module-transforms/src/index.ts

```diff
@@ -412,43 +412,92 @@ function buildExportInitializationStatements(
   constantReexports: boolean = false,
   noIncompleteNsImportDetection = false,
 ) {
-  const initStatements = [];
+  const initStatements: Array<[string, t.Statement | null]> = [];
 
-  const exportNames = [];
   for (const [localName, data] of metadata.local) {
     if (data.kind === "import") {
       // No-open since these are explicitly set with the "reexports" block.
     } else if (data.kind === "hoisted") {
-      initStatements.push(
+      initStatements.push([
+        // data.names is always of length 1 because a hoisted export
+        // name must be id of a function declaration
+        data.names[0],
         buildInitStatement(metadata, data.names, identifier(localName)),
-      );
-    } else {
-      exportNames.push(...data.names);
+      ]);
+    } else if (!noIncompleteNsImportDetection) {
+      for (const exportName of data.names) {
+        initStatements.push([exportName, null]);
+      }
     }
   }
 
   for (const data of metadata.source.values()) {
     if (!constantReexports) {
-      initStatements.push(...buildReexportsFromMeta(metadata, data, false));
+      const reexportsStatements = buildReexportsFromMeta(metadata, data, false);
+      const reexports = [...data.reexports.keys()];
+      for (let i = 0; i < reexportsStatements.length; i++) {
+        initStatements.push([reexports[i], reexportsStatements[i]]);
+      }
     }
-    for (const exportName of data.reexportNamespace) {
-      exportNames.push(exportName);
+    if (!noIncompleteNsImportDetection) {
+      for (const exportName of data.reexportNamespace) {
+        initStatements.push([exportName, null]);
+      }
     }
   }
 
-  if (!noIncompleteNsImportDetection) {
-    initStatements.push(
-      ...chunk(exportNames, 100).map(members => {
-        return buildInitStatement(
-          metadata,
-          members,
-          programPath.scope.buildUndefinedNode(),
+  // https://tc39.es/ecma262/#sec-module-namespace-exotic-objects
+  // The [Exports] list is ordered as if an Array of those String values
+  // had been sorted using %Array.prototype.sort% using undefined as comparefn
+  initStatements.sort((a, b) => (a[0] > b[0] ? 1 : -1));
+
+  const results = [];
+  if (noIncompleteNsImportDetection) {
+    for (const [, initStatement] of initStatements) {
+      results.push(initStatement);
+    }
+  } else {
+    // We generate init statements (`exports.a = exports.b = ... = void 0`)
+    // for every 100 exported names.
+    const chunkSize = 100;
```

Comment:
The chunk behaviour has existed since https://github.com/babel/babel/commit/95e08b6d2a16d2059284361e8aba8caf7338cfa8#diff-f9b022d801ca0bab9bbb7057b9ac49aac0d8c2ab53f41f97e7e15bdd10f26069R272. I don't know if it is a workaround of browser quirks or it is for aesthetic reasons.

Reply:
I think it's to avoid a too deeply nested AST

Reply:
Ref - https://github.com/babel/babel/issues/14272